bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/api/JuiceNetApi.java

```diff
@@ -0,0 +1,230 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.juicenet.internal.api;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiDevice;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiDeviceStatus;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiInfo;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiTouSchedule;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link JuiceNetApi} is responsible for implementing the api interface to the JuiceNet cloud server
+ *
+ * @author Jeff James - Initial contribution
+ */
+@NonNullByDefault
+public class JuiceNetApi {
+    private final Logger logger = LoggerFactory.getLogger(JuiceNetApi.class);
+
+    private static final String API_HOST = "https://jbv1-api.emotorwerks.com/";
+    private static final String API_ACCOUNT = API_HOST + "box_pin";
+    private static final String API_DEVICE = API_HOST + "box_api_secure";
+
+    protected String apiToken = "";
+    protected JuiceNetHttp httpApi;
+    @Nullable
+    protected ThingUID bridgeUID;
+
+    public JuiceNetApi(HttpClient httpClient) {
+        this.httpApi = new JuiceNetHttp(httpClient);
+    }
+
+    public void initialize(String apiToken, ThingUID bridgeUID) {
+        this.apiToken = apiToken;
+        this.bridgeUID = bridgeUID;
+    }
+
+    public List<JuiceNetApiDevice> queryDeviceList()
+            throws JuiceNetApiException, IOException, InterruptedException, TimeoutException, ExecutionException {
+        Map<String, Object> params = new HashMap<>();
+        ContentResponse response;
+
+        params.put("cmd", "get_account_units");
+        params.put("device_id", bridgeUID.getAsString());
+        params.put("account_token", apiToken);
+
+        response = httpApi.httpPost(API_ACCOUNT, params);
+
+        if (response.getStatus() != HttpStatus.OK_200) {
+            throw new JuiceNetApiException("Unable to retrieve device list, please check configuration (HTTP code: "
+                    + response.getStatus() + ").");
+        }
+
+        logger.trace("{}", response.getContentAsString());
```

Comment:
Some overhead could be avoided by storing content in local String as `getContentAsString()` as called again in next line.